vro-types/o11n-plugin-aria/index.d.ts

```diff
@@ -10,16 +10,429 @@ declare class VraHost {
+	constructor();
+	constructor();
+
+	/**
+	 * @param key
+	 * @param linksItem
+	 */
+	putLinksItem(key: string, linksItem: VraHref): VraCloudAccountVsphere;
+
+	/**
+	 * @param key
+	 * @param customPropertiesItem
+	 */
+	putCustomPropertiesItem(
+		key: string,
+		customPropertiesItem: string
+	): VraCloudAccountVsphere;
+}
+
+declare class VraCloudZoneService {
+	constructor();
+	constructor();
+
+	/**
+	 * @param genericRestClient
+	 */
+	setGenericRestClient(genericRestClient: VraGenericRestClient): void;
+
+	deleteZoneById(zoneId: string): void;
+
+	/**
+	 * This method deletes Zone based on given zone.
+	 * @param zone
+	 */
+	deleteZone(zone: VraZone): void;
+
+	/**
+	 * This method updates the Zone based on zoneId & ZoneSpecification.
+	 * @param id
+	 * @param body
+	 */
+	updateZone(id: string, body: VraZoneSpecification): VraZone;
+
+	/**
+	 * This method creates Zone based on ZoneSpecification details.
+	 * @param body
+	 */
+	createZone(body: VraZoneSpecification): VraZone;
+}
+
+declare class VraZone {
+	owner: string;
+	linksExtension: string;
+	externalRegionId: string;
+	cloudAccountId: string;
+	description: string;
+	orgId: string;
+	tags: Object[];
+	createdAt: string;
+	folder: string;
+	tagsToMatch: Object[];
+	internalIdString: string;
+	name: string;
+	host: VraHost;
+	placementPolicy: string;
+	tagsExtension: string;
+	tagsToMatchExtension: string;
+	id: string;
+	updatedAt: string;
+	customPropertiesExtension: string;
+	constructor();
+	constructor();
+
+	/**
+	 * @param tagsToMatchItem
+	 */
+	addTagsToMatchItem(tagsToMatchItem: VraTag): VraZone;
+
+	/**
+	 * @param key
+	 * @param linksItem
+	 */
+	putLinksItem(key: string, linksItem: VraHref): VraZone;
+
+	/**
+	 * @param tagsItem
+	 */
+	addTagsItem(tagsItem: VraTag): VraZone;
+
+	/**
+	 * @param key
+	 * @param customPropertiesItem
+	 */
+	putCustomPropertiesItem(key: string, customPropertiesItem: string): VraZone;
+}
+
+declare class VraHref {
+	hrefs: Object[];
+	href: string;
+	constructor();
+	constructor();
+
+	/**
+	 * @param hrefsItem
+	 */
+	addHrefsItem(hrefsItem: string): VraHref;
+}
+
+/**
+ * Specification for a zone.
+ */
+declare class VraZoneSpecification {
+	folder: string;
+	computeIds: Object[];
+	tagsToMatch: Object[];
+	regionId: string;
+	name: string;
+	placementPolicy: string;
+	description: string;
+	tags: Object[];
+	constructor();
+	constructor();
+
+	/**
+	 * @param tagsToMatchItem
+	 */
+	addTagsToMatchItem(tagsToMatchItem: VraTag): VraZoneSpecification;
+
+	/**
+	 * @param key
+	 * @param customPropertiesItem
+	 */
+	putCustomPropertiesItem(
+		key: string,
+		customPropertiesItem: string
+	): VraZoneSpecification;
 
+	/**
+	 * @param computeIdsItem
+	 */
+	addComputeIdsItem(computeIdsItem: string): VraZoneSpecification;
+
+	/**
+	 * @param tagsItem
+	 */
+	addTagsItem(tagsItem: VraTag): VraZoneSpecification;
+}
+
+declare class VraTag {
+	readonly displayName: string;
+	readonly internalIdString: string;
```

Comment:
Are those 2 properties intentionally left as readonly?

Reply:
yes, they are

![image](https://github.com/user-attachments/assets/6e424ab2-3bd8-45d2-a6c9-98590d3add08)

Reply:
Listed as such in the API